pkg/test/mocks/client.go

```diff
@@ -46,11 +46,44 @@ func (f FakeClient) Get(ctx context.Context, key client.ObjectKey, out client.Ob
 	return err
 }
 
-func (FakeClient) List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
-	panic("not implemented")
+func (f FakeClient) List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+	gvkUnprocessed, err := apiutil.GVKForObject(list, f.scheme)
+	if err != nil {
+		return err
+	}
+	// this will be a SomethingList kind, and we want to get the Something: remove `List` (last 4 chars) of this string
+	nonListKind := gvkUnprocessed.Kind
+	nonListKind = nonListKind[:len(nonListKind)-4]
+	gvk := schema.GroupVersionKind{gvkUnprocessed.Group, gvkUnprocessed.Version, nonListKind}
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
```

Comment:
Not super happy with the way of figuring out gvk and gvr here - any suggestions on how to make it nicer?

Reply:
+1 for this, I wonder if folks from api-server will be more aware about this.

Reply:
I'll take a look but I don't want to hold up this PR (sorry for the delay already!) :-)